sdk/synapse/azure-analytics-synapse-accesscontrol/src/main/java/com/azure/analytics/synapse/accesscontrol/AccessControlAsyncClient.java

```diff
@@ -0,0 +1,497 @@
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> deleteRoleAssignmentByIdWithResponse(String roleAssignmentId, Context context) {
+        return this.serviceClient.deleteRoleAssignmentByIdWithResponseAsync(roleAssignmentId, context);
+    }
+
+    /**
+     * Delete role assignment by role assignment Id.
+     * 
+     * @param roleAssignmentId The ID of the role assignment.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteRoleAssignmentById(String roleAssignmentId) {
+        return this.serviceClient.deleteRoleAssignmentByIdAsync(roleAssignmentId);
+    }
+
+    /**
+     * Delete role assignment by role assignment Id.
+     * 
+     * @param roleAssignmentId The ID of the role assignment.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteRoleAssignmentById(String roleAssignmentId, Context context) {
+        return this.serviceClient.deleteRoleAssignmentByIdAsync(roleAssignmentId, context);
+    }
+
+    /**
+     * List role assignments of the caller.
+     * 
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return array of Post200ApplicationJsonItemsItem.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SimpleResponse<List<String>>> getCallerRoleAssignmentsWithResponse() {
+        return this.serviceClient.getCallerRoleAssignmentsWithResponseAsync();
+    }
+
+    /**
+     * List role assignments of the caller.
+     * 
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return array of Post200ApplicationJsonItemsItem.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SimpleResponse<List<String>>> getCallerRoleAssignmentsWithResponse(Context context) {
```

Comment:
We do not need methods taking `Context` arguments in the async clients.

Reply:
Thanks. Will fix this.

Reply:
Fixed